pkgs/by-name/am/amber-lang/package.nix

```
@@ -0,0 +1,76 @@
+{ lib,
+  fetchFromGitHub,
+  rustPlatform,
+  bc,
+  bash,
+  makeWrapper,
+  runCommand,
+  amber-lang
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "amber-lang";
+  version = "0.3.1-alpha";
+
+  src = fetchFromGitHub {
+    owner = "Ph0enixKM";
+    repo = "Amber";
+    rev = version;
+    hash = "sha256-VSlLPgoi+KPnUQJEb6m0VZQVs1zkxEnfqs3fAp8m1o4=";
+  };
+
+  cargoHash = "sha256-NzcyX/1yeFcI80pNxx/OTkaI82qyQFJW8U0vPbqSU7g=";
+
+  nativeBuildInputs = [ bc bash ];
+  buildInputs = [ makeWrapper ];
+
+  preConfigure = ''
+    substituteInPlace src/compiler.rs \
+      --replace "/bin/bash" "${bash}/bin/bash"
```

Although the `${bash}` here is hostPlatform, considering that the compiled binary is also hostPlatform, and we cannot run checkPhase in cross compilation, this is not a big mistake.

btw
```suggestion
      --replace-fail "/bin/bash" "${bash}/bin/bash"
```

---

But

```suggestion
      --replace-fail "/bin/bash" "bash"
```

Would be enough